commands/createfixpullrequests_test.go

```diff
@@ -1,12 +1,100 @@
+	{commandArgs: []string{"up"}, technology: strings.ToLower(coreutils.Yarn), impactedPackaged: "minimist", fixVersion: "1.2.6", operator: "@", packageDescriptor: "package.json", fixPackageVersion: getGenericFixPackageVersionFunc()},
+	{commandArgs: []string{"install"}, technology: coreutils.Pipenv, impactedPackaged: "pyjwt", fixVersion: "2.4.0", operator: "==", packageDescriptor: "Pipfile", fixPackageVersion: getGenericFixPackageVersionFunc()},
+	{technology: coreutils.Maven, impactedPackaged: "junit", fixVersion: "4.11", packageDescriptor: "pom.xml", fixPackageVersion: getMavenFixPackageVersionFunc()},
+	{technology: coreutils.Pip, impactedPackaged: "pyjwt", fixVersion: "2.4.0", packageDescriptor: "requirements.txt", fixPackageVersion: getPipFixPackageVersionFunc()},
+	{technology: coreutils.Pip, impactedPackaged: "pyjwt", fixVersion: "2.4.0", packageDescriptor: "setup.py", fixPackageVersion: getPipFixPackageVersionFunc()},
+}
+
+var requirementsFile = "oslo.config>=1.12.1,<1.13\noslo.utils<5.0,>=4.0.0\nparamiko==2.7.2\npasslib<=1.7.4\nprance>=0.9.0\nprompt-toolkit~=1.0.15\npyinotify>0.9.6\nPyJWT>1.7.1\nurllib3 > 1.1.9, < 1.5.*"
+
+type pipPackageRegexTest struct {
+	packageName         string
+	expectedRequirement string
+}
+
+var pipPackagesRegexTests = []pipPackageRegexTest{
+	{"oslo.config", "oslo.config>=1.12.1,<1.13"},
+	{"oslo.utils", "oslo.utils<5.0,>=4.0.0"},
+	{"paramiko", "paramiko==2.7.2"},
+	{"passlib", "passlib<=1.7.4"},
+	{"prance", "prance>=0.9.0"},
+	{"prompt-toolkit", "prompt-toolkit~=1.0.15"},
+	{"pyinotify", "pyinotify>0.9.6"},
+	{"pyjwt", "PyJWT>1.7.1"},
+	{"urllib3", "urllib3 > 1.1.9, < 1.5.*"},
+}
+
+func getGenericFixPackageVersionFunc() FixGenericPackagesTestFunc {
+	return func(test packageFixTest) error {
+		return fixPackageVersionGeneric(test.commandArgs, test.technology, test.impactedPackaged, test.fixVersion, test.operator)
+	}
+}
+
+func getMavenFixPackageVersionFunc() FixMavenPackagesTestFunc {
+	return func(test packageFixTest) error {
+		mavenDepToPropertyMap := map[string][]string{
+			test.impactedPackaged: {"junit:junit", "3.8.1"},
+		}
+		cfp := CreateFixPullRequestsCmd{
+			mavenDepToPropertyMap: mavenDepToPropertyMap,
+		}
+		return fixPackageVersionMaven(&cfp, test.impactedPackaged, test.fixVersion)
+	}
+}
+
+func getPipFixPackageVersionFunc() FixPipPackagesTestFunc {
+	return func(test packageFixTest) error {
+		return fixPackageVersionPip(test.impactedPackaged, test.fixVersion, test.packageDescriptor)
+	}
+}
+
+func Test_fixPackageVersion(t *testing.T) {
+	testdataDir, err := filepath.Abs(filepath.Join("testdata/projects"))
+	assert.NoError(t, err)
+	for _, test := range packageFixTests {
+		// Create temp technology project
+		projectPath := filepath.Join(testdataDir, test.technology)
+		tmpProjectPath, cleanup := testdatautils.CreateTestProject(t, projectPath)
+		defer cleanup()
+		assert.NoError(t, os.Chdir(tmpProjectPath))
```

Comment:
We have a utility function that changes the dir, and also allows changing the dir back at the end.
You can sonsult with @yahavi about this.